src/API.jl

```diff
@@ -568,27 +568,47 @@ function status(ctx::Context, mode=PKGMODE_PROJECT)
     return
 end
 
-activate() = (Base.ACTIVE_PROJECT[] = Base.load_path_expand(nothing))
-function activate(path::String)
-    devpath = nothing
-    env = Base.active_project() === nothing ? nothing : EnvCache()
-    if env !== nothing && haskey(env.project["deps"], path)
-        uuid = UUID(env.project["deps"][path])
-        info = manifest_info(env, uuid)
-        devpath = haskey(info, "path") ? joinpath(dirname(env.project_file), info["path"]) : nothing
-    end
-    # `pkg> activate path`/`Pkg.activate(path)` does the following
-    # 1. if path exists, activate that
-    # 2. if path exists in deps, and the dep is deved, activate that path (`devpath` above)
-    # 3. activate the non-existing directory (e.g. as in `pkg> activate .` for initing a new env)
-    if Types.isdir_windows_workaround(path)
-        path = abspath(path)
-    elseif devpath !== nothing
-        path = abspath(devpath)
+activate() = (Base.ACTIVE_PROJECT[] = nothing)
+function activate(path::String; shared::Bool=false)
+    if !shared
+        devpath = nothing
+        env = Base.active_project() === nothing ? nothing : EnvCache()
+        if env !== nothing && haskey(env.project["deps"], path)
+            uuid = UUID(env.project["deps"][path])
+            info = manifest_info(env, uuid)
+            devpath = haskey(info, "path") ? joinpath(dirname(env.project_file), info["path"]) : nothing
+        end
+        # `pkg> activate path`/`Pkg.activate(path)` does the following
+        # 1. if path exists, activate that
+        # 2. if path exists in deps, and the dep is deved, activate that path (`devpath` above)
+        # 3. activate the non-existing directory (e.g. as in `pkg> activate .` for initing a new env)
+        if Types.isdir_windows_workaround(path)
+            fullpath = abspath(path)
+        elseif devpath !== nothing
+            fullpath = abspath(devpath)
+        else
+            fullpath = abspath(path)
+            isdir(fullpath) || @info("new environment will be placed at $fullpath")
+        end
     else
-        path = abspath(path)
+        # initialize `fullpath` in case of empty `Pkg.depots()`
+        fullpath = ""
+        # loop over all depots to check if the shared environment already exists
+        for depot in Pkg.depots()
+            fullpath = joinpath(Pkg.envdir(depot), path)
+            isdir(fullpath) && break
+        end
+        # this disallows names such as "Foo/bar", ".", "..", etc
+        if basename(abspath(fullpath)) != path
+            cmderror("not a valid name for a shared environment: $(path)")
+        end
+        # unless the shared environment already exists, place it in the first depots
+        if !isdir(fullpath)
+            fullpath = joinpath(Pkg.envdir(Pkg.depots1()), path)
+            @info("new shared environment \"$path\" will be placed at $fullpath")
```

Comment:
Should we warn here? We dont warn for `pkg> activate NonExistingDirectory`. I prefer not warning here I think.

Reply:
well i actually changed both to print an info for consistency. I could change it back though if thats prefered. My reasoning for printing an info is to catch typos more easily

Reply:
I prefer the `@info` warning, for the same reason as @Evizero. I don't think being too spammy will be an issue here, since creating a new environment should happen less often then activating an existing one.

Reply:
1. I think you will notice a typo, since you will notice that there is nothing there?
2. If we keep the `@info`, may I suggest that we print something alone the lines of 
```
[Info: activating new [shared] environment $path in $fullpath.
```

Reply:
Do we want to print `$path` even for non-shared? since then it could very well be something like `../Foo/Bar` which isn't much of a environment name

Reply:
We could print the full, normalized path for `../Foo/Bar`.

Reply:
In my experience, subtle issues like this are easy to miss when buried under other issues (or when you are a new user of an interface). Then you go: "Argh! I can't believe THAT was the issue!" and throw your computer out the window.

:+1:  @fredrikekre's ` @info` message

Reply:
We can tweak the exact way to info things in the upcoming PR but I aim to get the new commits into the rc3 so I merged this quickly.

Reply:
Yes we can later do a pass and make all the infos/warnings consistent with each other.